src/Kfz24/Client/Aws/LargePayloadMessageExtension.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Kfz24\QueueBundle\Client\Aws;
+
+use Aws\S3\S3Client;
+use Ramsey\Uuid\Uuid;
+
+class LargePayloadMessageExtension
+{
+    private const DEFAULT_LARGE_MESSAGE_SIZE_THRESHOLD = 250;
+    private const S3_BUCKET_NAME_MARKER = '-..s3BucketName..-';
+    private const S3_KEY_MARKER = '-..s3Key..-';
+
+    /**
+     * @var S3Client
+     */
+    private $s3Client;
+
+    /**
+     * @var string
+     */
+    private $bucketName;
+
+    /**
+     * @var int
+     */
+    private $largeMessageSizeThreshold;
+
+    /**
+     * @param S3Client $s3Client
+     * @param string $bucketName
+     */
+    public function __construct(S3Client $s3Client, string $bucketName)
+    {
+        $this->s3Client = $s3Client;
+        $this->bucketName = $bucketName;
+        $this->largeMessageSizeThreshold = self::DEFAULT_LARGE_MESSAGE_SIZE_THRESHOLD;
+    }
+
+    /**
+     * @return int
+     */
+    public function getLargeMessageSizeThreshold(): int
+    {
+        return $this->largeMessageSizeThreshold;
+    }
+
+    /**
+     * @param int $kilobytes
+     *
+     * @return LargePayloadMessageExtension
+     */
+    public function setLargeMessageSizeThreshold(int $kilobytes): LargePayloadMessageExtension
+    {
+        $this->largeMessageSizeThreshold = $kilobytes;
+
+        return $this;
+    }
+
+    /**
+     * @param array $message
+     *
+     * @return bool
+     */
+    public function isMessageLarge(array $message): bool
+    {
+        $messageSizeInBytes = strlen($message['MessageBody']);
+
+        if ($messageSizeInBytes < ($this->largeMessageSizeThreshold * 1024)) {
```

Comment:
Any reason to not use the same unit everywhere? (e.g: bytes).

Reply:
I implemented it with Kilobytes as the unit for users because AWS also specifies it in KB and it's easier for users to work with. Also fine tuning it to a specific byte amount is in my opinion not something that makes sense here.